sound/soc/sof/intel/hda-loader.c

```diff
@@ -472,6 +472,37 @@ int hda_dsp_post_fw_run(struct snd_sof_dev *sdev)
 	return hda_dsp_ctrl_clock_power_gating(sdev, true);
 }
 
+/* post fw run operations for ICL, C3 will be power up when HPRO */
+int hda_dsp_post_fw_run_icl(struct snd_sof_dev *sdev)
+{
+	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
+	int ret;
+
+	if (sdev->first_boot) {
+		ret = hda_sdw_startup(sdev);
+		if (ret < 0) {
+			dev_err(sdev->dev,
+				"error: could not startup SoundWire links\n");
+			return ret;
+		}
+	}
+
+	hda_sdw_int_enable(sdev, true);
+
+	if (!hda->clk_config_lpro) {
```

Comment:
Can we please add a comment here saying that the recoemmended HW programming sequence for IC>L is to power up core 3 and keep it in stall if HPRO is enabled